src/meta/processors/admin/Balancer.cpp

```diff
@@ -675,5 +794,84 @@ void Balancer::simplifyLeaderBalnacePlan(GraphSpaceID spaceId, LeaderBalancePlan
     }
 }
 
+bool Balancer::fetchHostLoadingBySpace(GraphSpaceID spaceId, int32_t replicaFactor) {
+    std::unique_ptr<kvstore::KVIterator> iter;
+    const auto& prefix = MetaServiceUtils::partPrefix(spaceId);
+    auto code = kv_->prefix(kDefaultSpaceId, kDefaultPartId, prefix, &iter);
+    if (code != kvstore::ResultCode::SUCCEEDED) {
+        LOG(ERROR) << "List Parts Failed";
+        return false;
+    }
+
+    std::unordered_map<HostAddr, int32_t> hostLoading;
+    while (iter->valid()) {
+        auto hosts = MetaServiceUtils::parsePartVal(iter->val());
+        for (auto& host : hosts) {
+            hostLoading[host]++;
```

Comment:
I don't think we should count for all hosts, only hosts in specified group should be counted.

Reply:
Not all hosts, it's only the space-related hosts